src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/MolecularFunctionAnnotationBuilder.java

```diff
@@ -21,18 +21,23 @@
      */
     public static List<String> processMolecularFunctions(GKInstance reactionInst) throws Exception {
         List<String> goaLines = new ArrayList<>();
-        Collection<GKInstance> catalystInstances = reactionInst.getAttributeValuesList(ReactomeJavaConstants.catalystActivity);
-        for (GKInstance catalystInst : catalystInstances) {
-            // Check that catalyst instance has no disqualifying attributes.
-            boolean validCatalyst = isValidMolecularFunctionCatalyst(catalystInst);
-            if (validCatalyst) {
-                // For MF annotations, gets all proteins from the ActiveUnit if it exists or from the PhysicalEntity if not.
-                GKInstance activeUnitInst = (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.activeUnit);
-                GKInstance entityInst = activeUnitInst == null ? (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.physicalEntity) : activeUnitInst;
-                Set<GKInstance> proteinInstances = getMolecularFunctionProteins(entityInst);
-                goaLines.addAll(processProteins(proteinInstances, reactionInst, catalystInst));
-            } else {
-                logger.info("Invalid catalyst, skipping GO annotation");
+        // As of v74 (September 2020), CatalystActivity literatureReferences are found in a ReactionlikeEvent's 'catalystActivityReference' attribute, not on the CA instance.
+        Collection<GKInstance> catalystReferenceInstances = reactionInst.getAttributeValuesList(ReactomeJavaConstants.catalystActivityReference);
+        for (GKInstance catalystReferenceInst : catalystReferenceInstances) {
+            // CatalystActivity instances are drawn from the CatalystActivityReference
+            Collection<GKInstance> catalystInstances = catalystReferenceInst.getAttributeValuesList(ReactomeJavaConstants.catalystActivity);
+            for (GKInstance catalystInst : catalystInstances) {
+                // Check that catalyst instance has no disqualifying attributes.
```

Comment:
I think `catalystActivity` is a single-valued attribute on `catalystActivityReference`, so you should be able to grab the single value in place of a loop.  I think there's a method on the Java Reactome API to do that?  If not, you could use an Optional<GKInstance> and the `ifPresent` method to call a function that handles the CA.  You could also use an if `statement` to check the CA isn't null.  Either way, it may be good to extract the logic in the for loop that handles the CA to its own method for clarity.